data/models.py

```diff
@@ -191,3 +191,123 @@ class LandoConnection(models.Model):
+    data_type = models.CharField(max_length=30, choices=JobQuestionDataType.ITEMS, blank=False, null=False,
+                                 help_text="Determines how answer is formatted in CWL input json.")
+    occurs = models.IntegerField(blank=False, null=False, default=1,
+                                 help_text="If > 1 this is an array in the CWL input json")
+
+    def __unicode__(self):
+        return '{} key:{} data_type: {}'.format(self.id, self.key, self.data_type)
+
+
+class JobQuestionnaire(models.Model):
+    """
+    List of JobQuestions that are for a particular workflow version.
+    Contains questions that will be used to create CWL input json and job fields.
+    """
+    description = models.CharField(max_length=255, blank=False, null=False,
+                                   help_text="User facing description")
+    workflow_version = models.ForeignKey(WorkflowVersion, on_delete=models.CASCADE, blank=False, null=False,
+                                         help_text="Workflow that this questionaire is for")
+    questions = models.ManyToManyField(JobQuestion,
+                                       help_text="Questions that are required to create a job")
+
+    def __unicode__(self):
+        return '{} desc:{}'.format(self.id, self.description)
+
+
+class JobAnswerKind(object):
+    """
+    Determines which type of JobAnswer goes with a JobQuestion
+    """
+    STRING = 'string'
+    DDS_FILE = 'dds_file'
+    ITEMS = (
+        (STRING, 'DukeDS File'),
+        (DDS_FILE, 'Text'),
+    )
+
+
+class JobAnswer(models.Model):
+    """
+    Answer to a JobQuestion.
+    """
+    question = models.ForeignKey(JobQuestion, on_delete=models.CASCADE, null=False, related_name='answers')
+    questionnaire = models.ForeignKey(JobQuestionnaire, on_delete=models.CASCADE, null=True, blank=True,
+                                      help_text='When null this is a user editable answer otherwise it is a system answer')
+    index = models.IntegerField(null=True, default=0,
+                                help_text='Used to order array answers')
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, null=False,
+                             help_text='User who created this job answer.')
+    kind = models.CharField(max_length=30, choices=JobAnswerKind.ITEMS,
+                            help_text="Determines child table associated with this answer.")
+
+    def __unicode__(self):
+        return '{} question:{} - index:{}'.format(self.id, self.question.name, self.index)
+
+
+class JobStringAnswer(models.Model):
+    """
+    String value associated with a JobAnswer.
+    """
+    answer = models.OneToOneField(JobAnswer, on_delete=models.CASCADE, primary_key=True, related_name='string_value')
+    value = models.CharField(max_length=255, blank=False, null=False)
+
+    def __unicode__(self):
+        return '{} question:{} - value:{}'.format(self.id, self.answer.question.name, self.value)
```

Comment:
By setting `primary_key=True`, the `id` field is not added, so this raises an exception when I try to create a JobStringAnswer through the Admin UI. `self.pk`should work.

> AttributeError at /admin/data/jobstringanswer/add/
> 'JobStringAnswer' object has no attribute 'id'

https://docs.djangoproject.com/en/1.10/ref/models/fields/#primary-key